Adding Status and other Role models

blairliikala opened the pull request:
Added initial Status model info with properties.

blairliikala committed:
Created Status model

ops-andy commented:
@blairliikala thanks for the contribution!

- [x]  @intoeetive can you review this to ensure that technically everything is good to go.
- [x]  @blairliikala can you please update the _toc.yml file as well to add this new page to the navigation. You'll add this around Line 671.

ops-andy committed:
added new status model page to the TOC

ops-andy committed:
removed ad-hoc model listing from page.

This didn't fit the style of the docs and models described in docs can be found in navigation

blairliikala committed:
New RoleGroup model

RoleGroup with properties and examples.

blairliikala committed:
Add RoleGroup model link

blairliikala committed:
Added Properties & Examples

blairliikala committed:
Added Properties

intoeetive commented on docs/development/models/member.md:
```diff
@@ -17,6 +17,78 @@ lang: php
+
+### Optional:
+#### `member_id`
+#### `group_id`
```
I believe `group_id` property needs not be in the docks. Or at least marked as deprecated. It is present for the sake of making legacy code work, but when writing new code, `role_id` should always be used

blairliikala committed:
Role model updates

Adding memberField and roleSettings model.
Fix format consistency issues.
Marked group_id as deprecated.

blairliikala committed:
Added deprecated, formatting

blairliikala committed:
Updates

blairliikala committed:
New Models, changes to old ones.

Added new Models, Added Keys, formatting changes to old models, updated examples.

blairliikala committed:
Added two template models.

blairliikala committed:
Typo fix

blairliikala committed:
Typo Fixes, added validation.

blairliikala committed:
Added validation

blairliikala committed:
Typo fixes, Added validation

blairliikala committed:
Typo Fixes

blairliikala committed:
Redid Examples.

blairliikala committed:
Redid Examples

blairliikala committed:
Typo fix.

blairliikala committed:
Redid examples, and typo fixes.

blairliikala committed:
Title typos.

blairliikala committed:
Model update.

blairliikala committed:
link update

blairliikala committed:
file name changes.

blairliikala committed:
Added Channel Fieds

blairliikala committed:
Typo Fix

blairliikala committed:
Added Examples

blairliikala committed:
Initial

blairliikala committed:
Description Fix

blairliikala committed:
Initial

blairliikala committed:
Adding File

blairliikala committed:
Typo Fix

blairliikala committed:
Removed native

blairliikala commented:
A few general questions here:
1. Should the build-in event methods be included in docs such as beforeSave, beforeValidate...etc?  How about getters and setters and validation?  Looking at the Member example that was there before I started, I would assume events would be simply listed, but get/set would be included?
2. Are the class names documented correctly?
3. Some properties are required when _creating_ a model, so I listed them under "Required" but I wonder if this might confuse people when _getting_ a model.  Thoughts?
4. I'm fairly stuck on understanding the custom field models.
5. In all the examples I include a validate step.  Should this stay in?
6. Should services be referenced?  For example, with Log models, there is a legacy service to add logs that works pretty well and has less code.  In these docs, should that be noted as a recommended way to add logs?
7. In some models when I was testing examples, the Site ID was required, but wasn't always listed as required in the model file.  Should this be noted in docs or filed as bugs?  If there is only one site, should Site ID always be included?
8. For organizing, should the files be organized into folders like they are in the code or keep it completely flat?
9. Are there any models _not_ to include?

blairliikala committed:
Removed on... Methods

blairliikala commented:
To elaborate on custom fields, I think having simple example for `relationships`, `grid` and `fluid` would be helpful.  However I couldn't quite figure this out.

robsonsobral commented:
Thank you so much, @blairliikala !

Can you give this a chance of merging, @intoeetive ? Better to have partial docs than outdated ones!

intoeetive commented on docs/development/models/role.md:
```diff
@@ -7,16 +7,27 @@ lang: php
+#### `short_name` Unique, max 50
+
+### Optional
+#### `role_id` Key, ini
```
should this say "int", not "ini"?

blairliikala replied:
Yep!

intoeetive commented on docs/development/models/working-with-models.md:
```diff
@@ -18,7 +18,3 @@ All ExpressionEngine models (both built-in and third-party) can be accessed and
 The model service allows to to [fetch model instances](development/services/model/fetching.md) and grab their [relationships](development/services/model/relationships.md) as well as perform changes.
 
 All models have a [set of methods](development/services/model/building-your-own.md#model-methods) available to get and set the data, however some models also define their own set of methods.
-
-#### [Member model](development/models/member.md)
-
-#### [Role model](development/models/role.md)
```
I don't think this needs to be removed from this page. Instead, should we add the links from TOC here - so people would have more ways to get to the page they need to?

blairliikala replied:
Andy had said to remove it.  I could see duplicate links not being needed, but either way is fine to me I suppose.

blairliikala replied:
It took me a little bit to combine the knowledge from the Model Service docs and the Models themselves when I was learning, so maybe putting those two closer together somehow would be helpful.  Maybe that intro page should be repurposed around that?

intoeetive replied:
Maybe. If you feel you have ideas what to do here, please go ahead in implement. Otherwise we can just leave as it is for now

blairliikala replied:
Probably leave as-is and might come back to it for now.

intoeetive commented on docs/_toc.yml:
```diff
@@ -665,10 +665,41 @@
+          href: development/models/template.md
+        - name: Template Group
+          href: development/models/template-group.md
+
```
Can we add "Model" to each name here for clarity? E.g. "Role Model", "Template Model" etc.

blairliikala replied:
Done!

intoeetive reviewed (changes requested):
Blair - you did fantastic job creating all this documentation, with real-world detailed examples! Sorry it took me so long to review. I am suggesting just couple of minor changes, and then we're good to merged and publish these!

blairliikala committed:
Added Model to TOC

And fixed the ini to int.
